src/m365/commands/request.ts

```diff
@@ -0,0 +1,163 @@
+import { AxiosRequestConfig, AxiosRequestHeaders } from 'axios';
+import auth, { Auth } from '../../Auth';
+import { Logger } from '../../cli';
+import Command from '../../Command';
+import GlobalOptions from '../../GlobalOptions';
+import request from '../../request';
+import commands from './commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  url: string;
+  method?: string;
+  resource?: string;
+  body?: string;
+  headers?: string;
+  accept?: string;
+  contentType?: string;
+}
+
+class RequestCommand extends Command {
+  public get name(): string {
+    return commands.REQUEST;
+  }
+
+  public get description(): string {
+    return 'Invoke a custom request at a Microsoft 365 API';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        method: args.options.method || 'get',
+        resource: args.options.resource,
```

Comment:
If some calls a custom API secured with AAD, this string could contain PII, so let's not track the string itself but rather its usage

Reply:
🤙